cmd/collect.go

```diff
@@ -30,82 +26,75 @@ var collectCmd = &cobra.Command{
-	collectCmd.PersistentFlags().BoolVar(&forceUpdate, "force-update", false, "set flag to force update data sent to SMD")
-	collectCmd.PersistentFlags().StringVar(&cacertPath, "cacert", "", "path to CA cert. (defaults to system CAs)")
+	collectCmd.PersistentFlags().StringVar(&host, "host", "", "Set the URI to the SMD root endpoint")
+	collectCmd.PersistentFlags().StringVar(&username, "username", "", "Set the BMC user")
+	collectCmd.PersistentFlags().StringVar(&password, "password", "", "Set the BMC password")
+	collectCmd.PersistentFlags().StringVar(&scheme, "scheme", "https", "Set the scheme used to query")
+	collectCmd.PersistentFlags().StringVar(&protocol, "protocol", "tcp", "Set the protocol used to query")
+	collectCmd.PersistentFlags().StringVarP(&outputPath, "output", "o", fmt.Sprintf("/tmp/%smagellan/inventory/", currentUser.Username+"/"), "Set the path to store collection data")
+	collectCmd.PersistentFlags().BoolVar(&forceUpdate, "force-update", false, "Set flag to force update data sent to SMD")
+	collectCmd.PersistentFlags().StringVar(&cacertPath, "cacert", "", "Path to CA cert. (defaults to system CAs)")
 
 	// set flags to only be used together
 	collectCmd.MarkFlagsRequiredTogether("username", "password")
 
 	// bind flags to config properties
-	viper.BindPFlag("collect.driver", collectCmd.Flags().Lookup("driver"))
-	viper.BindPFlag("collect.host", collectCmd.Flags().Lookup("host"))
-	viper.BindPFlag("collect.port", collectCmd.Flags().Lookup("port"))
-	viper.BindPFlag("collect.username", collectCmd.Flags().Lookup("username"))
-	viper.BindPFlag("collect.password", collectCmd.Flags().Lookup("password"))
-	viper.BindPFlag("collect.protocol", collectCmd.Flags().Lookup("protocol"))
-	viper.BindPFlag("collect.output", collectCmd.Flags().Lookup("output"))
-	viper.BindPFlag("collect.force-update", collectCmd.Flags().Lookup("force-update"))
-	viper.BindPFlag("collect.cacert", collectCmd.Flags().Lookup("secure-tls"))
-	viper.BindPFlags(collectCmd.Flags())
+	checkBindFlagError(viper.BindPFlag("collect.host", collectCmd.Flags().Lookup("collect.host")))
+	checkBindFlagError(viper.BindPFlag("collect.username", collectCmd.Flags().Lookup("collect.username")))
+	checkBindFlagError(viper.BindPFlag("collect.password", collectCmd.Flags().Lookup("collect.password")))
+	checkBindFlagError(viper.BindPFlag("collect.scheme", collectCmd.Flags().Lookup("collect.scheme")))
+	checkBindFlagError(viper.BindPFlag("collect.protocol", collectCmd.Flags().Lookup("collect.protocol")))
+	checkBindFlagError(viper.BindPFlag("collect.output", collectCmd.Flags().Lookup("collect.output")))
+	checkBindFlagError(viper.BindPFlag("collect.force-update", collectCmd.Flags().Lookup("collect.force-update")))
+	checkBindFlagError(viper.BindPFlag("collect.cacert", collectCmd.Flags().Lookup("collect.cacert")))
+	checkBindFlagError(viper.BindPFlags(collectCmd.Flags()))
```

Comment:
I'm getting bind flag errors when running any subcommand with `--help`:
```
# ./magellan collect --help
{"level":"error","error":"flag for \"collect.host\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"collect.username\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"collect.password\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"collect.scheme\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"collect.protocol\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"collect.output\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"collect.force-update\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"collect.cacert\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"concurrency\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"timeout\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"verbose\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"debug\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"access-token\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"cache\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
{"level":"error","error":"flag for \"update.protocol\" is nil","time":"2024-08-13T18:32:42Z","message":"failed to bind flag"}
[... help message follows ...]
```

Reply:
I think that's failing because of the `Lookup()`. I think each arg should just be something like `host` instead of `collect.host`.

Reply:
Seems like the `root` commands aren't binding for some reason.

Reply:
These errors are no longer appearing with [7285492](https://github.com/OpenCHAMI/magellan/pull/46/commits/7285492815c41e39c47fcc50bb78ab9465392b88).